src/core/exec-invoke.c

```diff
@@ -2877,6 +2976,18 @@ static int setup_exec_directory(
                         if (r < 0)
                                 goto fail;
                 }
+
+                /* Apply storage quotas if needed */
+                QuotaLimit ql;
+                ql = context->directories[type].exec_quota;
+
+                if (IN_SET(type, EXEC_DIRECTORY_STATE, EXEC_DIRECTORY_CACHE, EXEC_DIRECTORY_LOGS) && ql.quota_set) {
+                        r = apply_exec_quotas(target_dir, params->cgroup_path, type, ql);
+                        if (r < 0) {
+                                log_debug("Unable to set storage limits for %s", target_dir);
```

Comment:
if you received an error always use log_debug_errno() and pass it, and suffix with `: %m`